File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:         
        l = head
        r = head
        prev = None
        counter = 0
        while r != None:
            r.prev = prev
            prev = r
            r = r.next
            counter += 1
        r = prev
        
        while counter != counter % 2:
            if l.val != r.val:
                return False
            l = l.next
            r = r.prev
            counter -= 2           
        return True

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        vector<int> nums;
        ListNode* cur = head;
        while (cur != nullptr) {
            nums.push_back(cur->val);
            cur = cur->next;
        }

        int l = 0, r = nums.size() - 1;
        while (l < r) {
            if (nums[l] != nums[r]) {
                return false;
            }
            ++l;
            --r;
        }
        return true;

    }
};